internal/controller/api_to_config_test.go

```diff
@@ -902,11 +902,297 @@ func TestConversion(t *testing.T) {
+													Mode:     v1beta1.AllowRestricted,
+												},
+											},
+										},
+									},
+									VRF:      "vrf2",
+									Prefixes: []string{"2001:db9::/96"},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: &frr.Config{
+				Routers: []*frr.RouterConfig{
+					{
+						MyASN:    65010,
+						RouterID: "192.0.2.5",
+						Neighbors: []*frr.NeighborConfig{
+							{
+								IPFamily: ipfamily.IPv4,
+								Name:     "65011@192.0.2.6",
+								ASN:      65011,
+								Addr:     "192.0.2.6",
+								Port:     179,
+								Outgoing: frr.AllowedOut{
+									PrefixesV4: []frr.OutgoingFilter{
+										{
+											IPFamily: ipfamily.IPv4,
+											Prefix:   "192.0.3.1/32",
+										},
+										{
+											IPFamily: ipfamily.IPv4,
+											Prefix:   "192.0.3.2/32",
+										},
+									},
+									PrefixesV6: []frr.OutgoingFilter{},
+								},
+								Incoming: frr.AllowedIn{
+									PrefixesV4: []frr.IncomingFilter{},
+									PrefixesV6: []frr.IncomingFilter{},
+								},
+							},
+							{
+								IPFamily: ipfamily.IPv4,
+								Name:     "65012@192.0.2.7",
+								ASN:      65012,
+								Addr:     "192.0.2.7",
+								Port:     179,
+								Outgoing: frr.AllowedOut{
+									PrefixesV4: []frr.OutgoingFilter{
+										{
+											IPFamily:    ipfamily.IPv4,
+											Prefix:      "192.0.2.10/32",
+											Communities: []string{"10:100", "10:101"},
+											LocalPref:   200,
+										},
+										{
+											IPFamily:    ipfamily.IPv4,
+											Prefix:      "192.0.2.11/32",
+											Communities: []string{"10:101"},
+										},
+										{
+											IPFamily:    ipfamily.IPv4,
+											Prefix:      "192.0.3.20/32",
+											Communities: []string{"10:100"},
+										},
+										{
+											IPFamily:    ipfamily.IPv4,
+											Prefix:      "192.0.3.21/32",
+											Communities: []string{"10:101"},
+											LocalPref:   200,
+										},
+									},
+									PrefixesV6: []frr.OutgoingFilter{},
+								},
+								Incoming: frr.AllowedIn{
+									PrefixesV4: []frr.IncomingFilter{},
+									PrefixesV6: []frr.IncomingFilter{},
+								},
+							},
+						},
+						VRF:          "",
+						IPV4Prefixes: []string{"192.0.2.10/32", "192.0.2.11/32", "192.0.3.1/32", "192.0.3.2/32", "192.0.3.20/32", "192.0.3.21/32"},
+						IPV6Prefixes: []string{},
+					},
+					{
+						MyASN:    65013,
+						RouterID: "2001:db8::3",
+						Neighbors: []*frr.NeighborConfig{
+							{
+								IPFamily: ipfamily.IPv4,
+								Name:     "65017@192.0.2.7",
+								ASN:      65017,
+								Addr:     "192.0.2.7",
+								Port:     179,
+								Outgoing: frr.AllowedOut{
+									PrefixesV4: []frr.OutgoingFilter{
+										{
+											IPFamily: ipfamily.IPv4,
+											Prefix:   "192.0.2.5/32",
+										},
+									},
+									PrefixesV6: []frr.OutgoingFilter{},
+								},
+								Incoming: frr.AllowedIn{
```

Comment:
can we test merging incoming and outgoing coming from different configs?

Reply:
added